doc/code/qml_fourier.rst

```diff
@@ -4,6 +4,302 @@ qml.fourier
 This module contains tools for computing Fourier series representations of
 quantum circuits.
 
-.. automodapi:: pennylane.fourier
-    :include-all-objects:
-    :no-inheritance-diagram:
+.. currentmodule:: pennylane.fourier
+
+Overview
+--------
+
+Computing the Fourier representation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. autosummary::
+    :toctree: api
+
+    coefficients
+
+Visualization
+^^^^^^^^^^^^^
+
+The visualization functions require the ``matplotlib`` library to be installed.
+
+.. autosummary::
+    :toctree: api
+
+    visualize.bar
+    visualize.box
+    visualize.panel
+    visualize.radial_box
+    visualize.violin
+
+
+Fourier representation of quantum circuits
+------------------------------------------
+
+Consider a quantum circuit that depends on a parameter vector :math:`x` with
+length :math:`N`. The circuit involves application of some unitary operations
+:math:`U(x)`, and then measurement of an observable :math:`\hat{O}`.
+Analytically, the expectation value is
+
+.. math::
+
+   \langle \hat{O} \rangle = \langle 0 \vert U^\dagger (x) \hat{O} U(x) \vert 0\rangle = \langle
+   \psi(x) \vert \hat{O} \vert \psi (x)\rangle.
+
+This output is simply a function :math:`f(x) = \langle \psi(x) \vert \hat{O} \vert \psi
+(x)\rangle`. Notably, it is a periodic function of the parameters, and
+it can thus be expressed as a multidimensional Fourier series:
+
+.. math::
+
+    f(x) = \sum \limits_{n_1\in \Omega_1} \dots \sum \limits_{n_N \in \Omega_N}
+    c_{n_1,\dots, n_N} e^{-i x_1 n_1} \dots e^{-i x_N n_N},
+
+where :math:`n_i` are integer-valued frequencies, :math:`\Omega_i` are the set
+of available values for the integer frequencies, and the
+:math:`c_{n_1,\ldots,n_N}` are Fourier coefficients.
+
+As a simple example, consider ``simple_circuit`` below, which is a function of a
+single parameter.
+
+.. code::
+
+    import pennylane as qml
+    from pennylane import numpy as np
+
+    dev = qml.device('default.qubit', wires=2)
+
+    @qml.qnode(dev)
+    def simple_circuit(x):
+        qml.RX(x[0], wires=0)
+        qml.RY(x[0], wires=1)
```

Comment:
Should this just be `x`, not `x[0]`? Or does the Fourier module require that `x` be a list?

Reply:
Nevermind, I read further and saw that it does require the input be a list :)